Source/RapyutaSimulationPlugins/Public/Core/RRCoreUtils.h

```diff
@@ -613,6 +617,31 @@ class RAPYUTASIMULATIONPLUGINS_API URRCoreUtils : public UBlueprintFunctionLibra
         InWorld->GetTimerManager().ClearTimer(InTimerHandle);
     }
 
+    // -------------------------------------------------------------------------------------------------------------------------
+    // PROCESS UTILS --
+    //
+    static int32 RunMonitoredProcess(FMonitoredProcess* InProcess, const float InTimeOutSecs)
+    {
+        // Launch [InProcess]
+        if (false == InProcess->Launch())
+        {
+            UE_LOG_WITH_INFO_SHORT(LogTemp, Error, TEXT("Failed launching process"));
+            return -1;
+        }
+
+        // Wait for it to finish with [InTimeOutSecs]
+        const float lastMarkedTime = URRCoreUtils::GetSeconds();
+        while (InProcess->Update())
+        {
+            // Already slept in [Update()]
+            if (URRCoreUtils::GetElapsedTimeSecs(lastMarkedTime) > InTimeOutSecs)
+            {
+                InProcess->Cancel();

```

Comment:
Don't you need break while loop or Update() will return false?

Reply:
Yeah I thought about it but then figured out the api actually already handles it more properly to make sure the while loop here only breaks after the process is terminated, specifically
* [Cancel() ](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Misc/MonitoredProcess.h#L71)-> ON `Canceling`
* [FMonitoredProcess::TickInternal()](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Private/Misc/MonitoredProcess.cpp#L135) -> upon cancelling turns OFF `bIsRunning` (which is of atomic type thus thread-safe) **only after officially terminating the process**, also making our while loop here break accordingly